Tests/DataPipeline/Core/UnitTest.swift

```diff
@@ -0,0 +1,81 @@
+@testable import CioDataPipelines
+@testable import CioInternalCommon
+import Foundation
+@testable import Segment
+import SharedTests
+
+/// Base class for unit testing within the module, extending `UnitTestBase` with setup and utilities
+/// specific to module components. Ideal for isolated tests of individual functions and classes.
+open class UnitTest: SharedTests.UnitTestBase<CustomerIO> {
+    public var dataPipelineConfigOptions: DataPipelineConfigOptions!
+    public var enableLogs: Bool = false
+
+    // Use this `CustomerIO` instance when invoking `CustomerIOInstance` functions in unit tests.
+    // This ensures convenience and consistency across unit tests, and guarantees the correct instance is used for testing.
+    var customerIO: CustomerIO!
+    var analytics: Analytics!
+
+    override open func setUp() {
+        setUp(modifyModuleConfig: nil)
+    }
+
+    override open func setUp(enableLogs: Bool = false, siteId: String? = nil, modifySdkConfig: ((inout SdkConfig) -> Void)?) {
+        setUp(enableLogs: enableLogs, siteId: siteId, modifySdkConfig: modifySdkConfig, modifyModuleConfig: nil)
+    }
+
+    open func setUp(
+        enableLogs: Bool = false,
+        siteId: String? = nil,
+        writeKey: String? = nil,
+        modifySdkConfig: ((inout SdkConfig) -> Void)? = nil,
```

Comment:
we should ideally be moving away from sdkConfig as well from test, maybe in next tickets.